src/main/java/pvs/app/service/GithubApiService.java

```diff
@@ -99,31 +106,43 @@ public boolean getCommitsFromGithub(String owner, String name, Date lastUpdate)
         Optional<JsonNode> paginationInfo = Optional.ofNullable(mapper.readTree(responseJson))
                 .map(resp -> resp.get("data"))
                 .map(data -> data.get("repository"))
-                .map(repo -> repo.get("defaultBranchRef"))
-                .map(branch -> branch.get("target"))
-                .map(tag -> tag.get("history"));
+                .map(repo -> repo.get("refs"))
+                .map(ref -> ref.get("edges"));
 
         if (paginationInfo.isPresent()) {
-            double totalCount = paginationInfo.get().get("totalCount").asInt();
-            List<GithubCommitLoaderThread> githubCommitLoaderThreadList = new ArrayList<>();
-
-            if (totalCount != 0) {
-                String cursor = paginationInfo.get().get("pageInfo").get("startCursor").textValue()
-                        .split(" ")[0];
-                for (int i = 1; i <= Math.ceil(totalCount / 100); i++) {
-                    GithubCommitLoaderThread githubCommitLoaderThread =
-                            new GithubCommitLoaderThread(
-                                    this.webClient,
-                                    this.githubCommitService,
-                                    owner,
-                                    name,
-                                    cursor + " " + (i * 100));
-                    githubCommitLoaderThreadList.add(githubCommitLoaderThread);
-                    githubCommitLoaderThread.start();
-                }
-
-                for (GithubCommitLoaderThread thread : githubCommitLoaderThreadList) {
-                    thread.join();
+            for (JsonNode objNode: paginationInfo.get()) {
+                Optional<JsonNode> branchName = Optional.ofNullable(objNode)
+                        .map(branch -> branch.get("node"))
+                        .map(node -> node.get("name"));
+                Optional<JsonNode> commitsFromABranch = Optional.ofNullable(objNode)
+                        .map(branch -> branch.get("node"))
+                        .map(node -> node.get("target"))
+                        .map(target -> target.get("history"));
+
+                if(branchName.isPresent() && commitsFromABranch.isPresent()) {
+                    double totalCount = commitsFromABranch.get().get("totalCount").asInt();
```

Comment:
Since there's exist a `asInt()` in the end of the statement, the type of `totalCount` can be set to `int`.
```suggestion
                    final int totalCount = commitsFromABranch.get().get("totalCount").asInt();
```